provers/risc0/agent/src/boundless.rs

```diff
@@ -439,163 +584,738 @@ impl Risc0BoundlessProver {
+                            requests_guard.insert(existing_request.request_id.clone(), existing_request.clone());
+                        }
+                    }
+                    return Ok(existing_request.request_id);
+                },
+                ProofRequestStatus::Locked { .. } => {
+                    tracing::info!("Returning existing locked aggregation proof (being processed by prover) for request: {}", existing_request.request_id);
+                    // Add to memory cache if not already there
+                    {
+                        let mut requests_guard = self.active_requests.write().await;
+                        if !requests_guard.contains_key(&existing_request.request_id) {
+                            requests_guard.insert(existing_request.request_id.clone(), existing_request.clone());
+                        }
+                    }
+                    return Ok(existing_request.request_id);
+                },
+                ProofRequestStatus::Failed { error } => {
+                    tracing::info!("Found failed request for same input ({}), creating new aggregation request", error);
+                    // Continue to create new request (allows retry)
+                }
+            }
         }
 
-        let (_, journal, seal) = self
-            .submit_and_wait_for_fulfillment(&boundless_client, request)
-            .await?;
-
-        let Ok(ContractReceipt::Base(boundless_receipt)) =
-            risc0_ethereum_contracts_boundless::receipt::decode_seal(
-                seal.clone(),
-                BOUNDLESS_BATCH_ID,
-                journal.clone(),
-            )
-        else {
-            return Err(AgentError::InvalidReceiptError);
+        // Prepare and store the async request
+        let request_id = self.prepare_async_request(
+            request_id,
+            ProofType::Aggregate,
+            input.clone(),
+            config,
+        ).await?;
+
+        // Submit to boundless market in background
+        let prover_clone = self.clone();
+        let active_requests = self.active_requests.clone();
+        let request_id_clone = request_id.clone();
+
+        tokio::spawn(async move {
+            let offer_params = prover_clone.boundless_config.get_aggregation_offer_params();
+            let image_url = prover_clone.aggregation_image_url.read().await.clone().unwrap();
+
+            if let Err(e) = prover_clone.process_and_submit_request(
+                &request_id_clone,
+                input,
+                BOUNDLESS_AGGREGATION_ELF,
+                image_url,
+                offer_params,
+                active_requests,
+            ).await {
+                prover_clone.update_failed_status(&request_id_clone, e.to_string()).await;
+            }
+        });
+
+        Ok(request_id)
+    }
+
+    /// Submit an update proof request asynchronously
+    pub async fn update(
```

Comment:
mark this one as todo, as for now we can skip support update.

meanwhile the comment is wrong.